pubsub/Readme.md

```diff
@@ -25,3 +25,44 @@ type PubSub interface {
 	Subscribe(req SubscribeRequest, handler func(msg *NewMessage) error) error
 }
 ```
+
+### TTL (or Time To Live)
+
+Time to live is implemented by default in Dapr. A publishing application can add the `expiration` attribute to a CloudEvent object and Dapr runtime will drop the message if the expiration is due at the subscriber's end, prior to invoking subscriber applications. The `expiration` attribute is Dapr specific and extends the CloudEvent spec. This is a convenience to subscriber applications, so Dapr does not expire messages from pub subs that do not have subscribers via Dapr.
```

Comment:
> A publishing application can add the `expiration` attribute to a CloudEvent object and Dapr runtime will drop the message if the expiration is due at the subscriber's end, prior to invoking subscriber applications. The `expiration` attribute is Dapr specific and extends the CloudEvent spec.

Where is this documented for **applications**? IIRC - Applications don't use cloud events to *send* pub/sub messages - they receive cloud events as the when handling a pub/sub message.

Reply:
@rynowak Actually, Dapr supports both. If the payload is a CloudEvent, Dapr will not double wrap it. There was an issue about that a while ago.

Reply:
> Dapr will not double wrap it.

It does remove the `time` attribute though, dunno if that's a bug.

Reply:
I was fixing that in this PR too but changed my mind to keep the scope within the issue here. I do believe that is a bug. I will create an issue for that.

Reply:
https://github.com/dapr/dapr/issues/2611